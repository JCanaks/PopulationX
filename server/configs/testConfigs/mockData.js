export default {
  signupInput: {
    validUserInput: {
      name: 'TestUser Signup',
      email: 'jsignup@mail.com',
      password: '1111',
    },
    invalidUserInput: {
      name: '',
      email: '',
      password: '',
    },
  },
  loginInput: {
    validUserInput: {
      email: 'jsignup@mail.com',
      password: '1111',
    },
    invalidUserInput: {
      email: '',
      password: '',
    },
    invalidPassword: {
      name: 'TestUser Signup',
      email: 'jsignup@mail.com',
      password: '1112',
    },
    invalidUser: {
      name: 'Invalid User',
      email: 'invalid@mail.com',
      password: '1112',
    },
  },
  locationInput: {
    validCreateLocationInput: {
      name: 'Lagos',
      noOfFemales: 1000,
      noOfMales: 950,
    },
    invalidCreateLocationInput: {
      name: '',
      locatedIn: '',
    },
    invalidCreateLocationInput1: {
      name: 'LLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLL',
      locatedIn: 'LLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLL',
    },
    invalidCreateLocationInput2: {
      name: 'L',
      locatedIn: 'L',
    },
    validUpdateLocationInput: {
      name: 'Lasgidi',
      noOfFemales: 2000,
      noOfMales: 1950,
    },
    invalidUpdateLocationInput: {
      name: 'L',
      locatedIn: 'n',
    },
    invalidUpdateLocationInput1: {
      name: 'LLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLL',
      locatedIn: 'LLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLL',
    },
  },
};
