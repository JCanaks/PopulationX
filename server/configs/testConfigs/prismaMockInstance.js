import { makePrismaClientClass } from 'prisma-client-lib';
import { typeDefs } from '../../generated/prisma-client/prisma-schema';
import { models } from '../../generated/prisma-client';

const PrismaMockInstance = makePrismaClientClass({
  typeDefs,
  models,
  endpoint: 'http://localhost:4499',
});
export default new PrismaMockInstance();
