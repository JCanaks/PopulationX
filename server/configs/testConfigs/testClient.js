import { ApolloServer, makeExecutableSchema } from 'apollo-server-express';
import { applyMiddleware } from 'graphql-middleware';
import { createTestClient } from 'apollo-server-testing';
import { resolvers, typeDefs, middlewares } from '../../graphqlServer';
import prismaMockInstance from './prismaMockInstance';

const createGraphqlTestClient = () => {
  const schema = makeExecutableSchema({
    typeDefs,
    resolvers,
  });
  const schemaWithMiddleware = applyMiddleware(schema, ...middlewares);
  const testServer = new ApolloServer({
    schema: schemaWithMiddleware,
    context: request => ({
      ...request,
      prisma: prismaMockInstance,
    }),
  });

  return createTestClient(testServer);
};

export default createGraphqlTestClient;
