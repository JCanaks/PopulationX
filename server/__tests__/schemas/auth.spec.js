import '@babel/polyfill';
import prismaMockInstance from '../../configs/testConfigs/prismaMockInstance';
import mockData from '../../configs/testConfigs/mockData';
import clearData from '../../utils/helpers/testHelpers';
import createGraphqlTestServer from '../../configs/testConfigs/testClient';
import { SIGNUP_USER, LOGIN_USER } from '../../configs/testConfigs/testConstants';

const { signupInput, loginInput } = mockData;
const { mutate } = createGraphqlTestServer();

describe('Testing auth resolvers', () => {
  beforeAll(async () => {
    await clearData(prismaMockInstance);
  });

  afterAll(async () => {
    await clearData(prismaMockInstance);
  });


  test('Signup with valid user input', async () => {
    const args = signupInput.validUserInput;
    const { errors, data } = await mutate({
      mutation: SIGNUP_USER,
      variables: args,
    });

    expect(data).toBeDefined();
    expect(data).toMatchSnapshot();
    expect(errors).toBeUndefined();
  });

  test('Signup with invalid user input', async () => {
    const args = signupInput.invalidUserInput;
    const { errors } = await mutate({
      mutation: SIGNUP_USER,
      variables: args,
    });
    
    expect(errors).toBeDefined();
    expect(errors).toMatchSnapshot();
  });

  test('Login with valid user input', async () => {
    const args = loginInput.validUserInput;
    const { errors, data } = await mutate({
      mutation: LOGIN_USER,
      variables: args,
    });

    expect(data).toBeDefined();
    expect(data).toMatchSnapshot();
    expect(errors).toBeUndefined();
  });

  test('Login with invalid user input', async () => {
    const args = loginInput.invalidUserInput;
    const { errors } = await mutate({
      mutation: LOGIN_USER,
      variables: args,
    });
    
    expect(errors).toBeDefined();
    expect(errors).toMatchSnapshot();
  });
});
