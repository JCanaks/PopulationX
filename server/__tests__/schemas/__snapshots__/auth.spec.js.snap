// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing auth resolvers Login with an invalid password 1`] = `
Array [
  [GraphQLError: Invalid Password],
]
`;

exports[`Testing auth resolvers Login with invalid user information 1`] = `
Array [
  [GraphQLError: No such user found],
]
`;

exports[`Testing auth resolvers Login with invalid user input 1`] = `
Array [
  [GraphQLError: {"email":["Please enter an email in the specified field"],"password":["Please enter a password in the specified field"]}],
]
`;

exports[`Testing auth resolvers Login with valid user input 1`] = `
Object {
  "login": Object {
    "user": Object {
      "email": "jsignup@mail.com",
      "name": "TestUser Signup",
    },
  },
}
`;

exports[`Testing auth resolvers Signup with invalid user input 1`] = `
Array [
  [GraphQLError: {"name":["Please enter a name in the specified field"],"email":["Please enter an email in the specified field"],"password":["Please enter a password in the specified field"]}],
]
`;

exports[`Testing auth resolvers Signup with valid user input 1`] = `
Object {
  "signup": Object {
    "user": Object {
      "email": "jsignup@mail.com",
      "name": "TestUser Signup",
    },
  },
}
`;
